include/swift/AST/Module.h

```diff
@@ -209,7 +210,20 @@ class ModuleDecl : public DeclContext, public TypeDecl {
 
 public:
   typedef ArrayRef<Located<Identifier>> AccessPathTy;
-  typedef std::pair<ModuleDecl::AccessPathTy, ModuleDecl*> ImportedModule;
+  
+  typedef struct _ImportedModule {
+    /// neither of these can be const because some things update after construction
+    ModuleDecl::AccessPathTy accessPathTy;
```

Comment:
Could you rename this field to `accessPath` (no `Ty`)? It seems counter-intuitive to have `Ty` in the name, since it is a field, not a type.

Reply:
Resolved in af9a745